server/auth/create_token.go

```diff
@@ -0,0 +1,96 @@
+package auth
+
+import (
+	"encoding/json"
+	"github.com/gmo-personal/coding_challenge/server/csrf"
+	"github.com/gmo-personal/coding_challenge/server/utils"
+	"time"
+
+	"github.com/gmo-personal/coding_challenge/database"
+	"github.com/gmo-personal/coding_challenge/model"
+	"golang.org/x/crypto/bcrypt"
+	"net/http"
+)
+
+// Structure of a session, containing account ID and expiry time.
+type Session struct {
+	AccountID string
+	Expires   time.Time
+}
+
+// In memory sessions table to keep track of active sessions.
+// TODO This should be a database table to avoid memory and concurrency issues as well as scaling.
+var Sessions = make(map[string]Session)
+
+// Generates auth and CSRF token and returns them to the client.
+func PostTokenHandler(w http.ResponseWriter, r *http.Request) {
+	// Attempts to read the body, if unable to do so, return bad request.
+	accountJson, err := utils.GetBody(r)
+	if err != nil {
+			utils.LogError(err)
+			utils.ServeBadRequest(w)
+			return
+	}
+
+	// Attempts to parse json into account model, if unable to do so, return bad request.
+	account := &model.Account{}
+	err = json.Unmarshal(accountJson, account)
+	if err != nil {
+			utils.LogError(err)
+			utils.ServeBadRequest(w)
+			return
+	}
+
+	// Attempts to find an existing account by username and compares input and existing passwords,
+	// Serves unauthorized if they do not match.
+	existingAccount, err := database.SelectAccountByUsername(account.Username)
+	if err != nil {
+			utils.LogError(w)
+			utils.ServeNotFound(w)
```

Comment:
returning NotFound here can be used to enumerate registered users
```suggestion
			utils.ServeUnauthorized(w)
```